src/main/java/codeu/model/data/Event.java

```diff
@@ -26,54 +22,27 @@
    * NOTE: At minimum, must store Username
    */
   private final List<String> information;
-  private UserStore userStore;
-  private ConversationStore conversationStore;
   
   /**
    * Constructs a new Event
    * 
-   * @param type the type of event
-   * @param creation the time the event was created
-   * @param classType the class of the event 
+   * @param id the ID of this Event
+   * @param type the type of this Event
+   * @param creation the creation time of this Event
+   * @param information the information of this Event 
    */
-  public Event(String type, Instant creation, Object classType) {
+  public Event(UUID id, String type, Instant creation, List<String> information) {
+    this.id = id;
     this.type = type;
     this.creation = creation;
-    information = new ArrayList<>();
-    setUserStore(userStore.getInstance());
-    setConversationStore(conversationStore.getInstance());
-
-    if (type.equals("User")) {
-      User user = (User)classType;
-      information.add(user.getName());
-    }
-    else if (type.equals("Conversation")) {
-      Conversation conversation = (Conversation)classType;
-      information.add(userStore.getUser(conversation.getOwnerId()).getName());
-      information.add(conversation.getTitle());
-    }
-    else if (type.equals("Message")) {
-      Message message = (Message)classType;
-      information.add(userStore.getUser(message.getAuthorId()).getName());
-      information.add(conversationStore.getConversation(message.getConversationId()).getTitle());
-      information.add(message.getContent());
-    }
+    this.information = new ArrayList<>();
+    for (String entry : information)
```

Comment:
Isn't this equivalent to just setting `this.information = information`?

Reply:
I did it this way thinking of shallow vs. deep copy, but I may have been wrong. Should I change it?

Reply:
But do we need a deep copy? Do you plan for the caller of the constructor to modify the List object after passing it in?